tutorials/cloud-run-with-pomerium-for-end-user-access/index.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Cloud Run with Pomerium for end-user access
+description: This guide covers how to deploy Pomerium to Cloud Run and use it to protect other endpoints via Authorization Headers.
+author: desimone
+tags: Cloud Run, Pomerium
+date_published: 2020-08-13
+---
+
+# Cloud Run with Pomerium for end-user access
+
+This guide covers how to deploy Pomerium to Cloud Run to provide end-user auth to other endpoints. 
+
+[Pomerium](https://www.pomerium.com) is an open source identity-aware proxy that enables secure access to internal applications. Pomerium provides a standardized interface to add access control to applications regardless of whether the application itself has authorization or authentication baked-in. Pomerium gateways  requests and can be used in situations where you'd typically reach for a VPN.
+
+## How it works
+
+Services on [Cloud Run](https://cloud.google.com/run) and other Google Cloud serverless products can be restricted to only permit access with a properly signed [bearer token](https://cloud.google.com/run/docs/authenticating/service-to-service). This allows requests from other services running in GCP or elsewhere to be securely authorized despite the endpoints being public.
+
+These bearer tokens are not easily set in a browser session and must be refreshed on a regular basis, preventing them from being useful for end user authorization. Pomerium, however, can generate compatible tokens on behalf of end users and proxy the request to these services.
```

Comment:
> These bearer tokens are not easily set in a browser session and must be refreshed on a regular basis, preventing them from being useful for end user authorization.

easily would be an understatement for "impossible" :)

you can be bold and say:

> These identity tokens can be generated in the browser, however browsers don't support adding these tokens to the `Authorization` header. Therefore, to authenticate end users to your Cloud Run services, you need a proxy to perform this for you.

Reply:
How about this? 

```suggestion
Browsers do not support adding these identity tokens to the `Authorization` header.  Therefore, a proxy is required to authenticate end users to your Cloud Run service.  Pomerium can perform this role by generating compatible tokens on behalf of end users and then proxy their requests.
```

Reply:
Based on Travis's suggestion, I added this:
"Browsers are not able to add these identity tokens to the `Authorization` header, so a proxy is required to authenticate end users to your Cloud Run service. Pomerium can perform this role by generating compatible tokens on behalf of end users and then acting as a proxy for their requests."